### PR TITLE
Add node20 to runner

### PR DESCRIPTION
**Pull request by takost**
Add node20 to runner
Part of:https://github.com/github/c2c-actions/issues/5943

### CODE REVIEW

**Review by TingluoHuang — APPROVED** [review]

**Review by D5688 — COMMENTED** [review]
@cdb 

- [ ]

File: src/Misc/externals.sh [hunk 1]
```diff
@@ -158,23 +163,29 @@ fi
 # Download the external tools only for OSX.
 if [[ "$PACKAGERUNTIME" == "osx-x64" ]]; then
     acquireExternalTool "$NODE_URL/v${NODE16_VERSION}/node-v${NODE16_VERSION}-darwin-x64.tar.gz" node16 fix_nested_dir
+    acquireExternalTool "$NODE_URL/v${NODE20_VERSION}/node-v${NODE20_VERSION}-darwin-x64.tar.gz" node20 fix_nested_dir
```
Comment (Guasonve) [inline src/Misc/externals.sh:166]: Mio

File: src/Misc/externals.sh [hunk 2]
```diff
@@ -158,23 +163,29 @@ fi
...
 fi
 
 if [[ "$PACKAGERUNTIME" == "osx-arm64" ]]; then
     # node.js v12 doesn't support macOS on arm64.
     acquireExternalTool "$NODE_URL/v${NODE16_VERSION}/node-v${NODE16_VERSION}-darwin-arm64.tar.gz" node16 fix_nested_dir
+    acquireExternalTool "$NODE_URL/v${NODE20_VERSION}/node-v${NODE20_VERSION}-darwin-arm64.tar.gz" node20 fix_nested_dir
```
Comment (Guasonve) [inline src/Misc/externals.sh:172]: Pa

File: src/Misc/externals.sh (hunk 1 above)
Comment (Guasonve) [inline src/Misc/externals.sh:166]: Vgf

**Comment by iamstarkov** [comment]
@takost @D5688 @Guasonve do you know if we can expect node24 support anytime soon?

**Pull request merged**